### PR TITLE
Remove unused NoopScope from context

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

**Comment by carlosalberto** [comment]
Ooops, somehow I lost track of this one ;(

**Comment by iNikem** [comment]
@bogdandrutu This class is still used by https://github.com/open-telemetry/opentelemetry-auto-instr-java/blob/1f0cd54fd745fd5c914284c9a3f583faff65dddc/instrumentation/opentelemetry-api-0.3/src/main/java/io/opentelemetry/auto/instrumentation/opentelemetryapi/context/ContextUtils.java

So  atm auto-instr does not compile against current snapshot of otel-java

**Comment by jkwatson** [comment]
@iNikem If you need a no-op Scope, the auto-instrumentation project can easily create one, since `Scope` is an interface.

**Comment by iNikem** [comment]
@jkwatson I don't need it. But auto-instr apparently does. My concern is expressed in https://github.com/open-telemetry/opentelemetry-auto-instr-java/issues/363

**Comment by jkwatson** [comment]
Sorry, let me rephrase... if the auto-instrumentation project needs it, you can easily create on there.
